### DISCUSSION TITLE
Edges broken when scaling flow pane

### DISCUSSION THREAD

**Discussion by elringus** [xyflow/xyflow#3229] [Q&A]
When applying `transform: scale()` to the flow container/pane the edges are positioned with offset:

https://github.com/wbkd/react-flow/assets/2056864/4da3a114-5ea6-4f16-b8fa-424e0d78c4a1

I guess this happens because the edge SVGs are drawn before the scale animation completes. Is there a way to draw them so that parent elements scale doesn't affect their positions? (I'm aware of react-flow's built-in zoom func, but it doesn't work for me, as I need custom easing and timing for translation and zoom)

**Comment by elringus** [comment]
Or more like I guess the edge positions are evaluated with some kind of delay after `setEdges` is invoked. When I start the animation with a delay they are resolved correctly. Is there a way to force edges initialization/positioning in a blocking manner, so I can start animation right after setting/changing them?

**Comment by elringus** [comment]
I've created a minimal repro and started an issue: https://github.com/wbkd/react-flow/issues/3231

**Reply by bcakmakoglu** [reply]
This is a known issue and tbh, the only workaround I know is either delay rendering your graph until the animation is done *or don't use a scaling animation at all*.
If it's about some modal popup opening with that sort of animation, consider using a fade instead.

I'm pretty sure there's no fix in sight for this, but I might be wrong ^^

**Reply by elringus** [reply]
Hey, Thanks for the insight! If there is no fix, we should make one :) I can't believe something can't be fixed (unless it's by-design, which is not the case here).

**Reply by bcakmakoglu** [reply]
If I remember correctly the issue here is something like this:

When you use a scaling animation, or really any animation that transforms the size of the graph, the moment the graph is rendered, that transformation causes the calculation for handle bounds to not be "accurate" (done using `getBoundingClientRect`).
Technically it *is accurate* but the problem is that after the animation has ended (which could be after an arbitrary time React Flow doesn't really know about), the graph is not the same size or whatever the transformation did to it.
Now to solve this, you could use `useUpdateNodeInternals` to force an update on these values and let React Flow know about that, or you can hide the graph until the animation has ended, or just not use it.

You could try to come up with a solution but like I said, this isn't a new issue and it hasn't been fixed yet - I tried it myself as well and gave up after a while as I did not find a solution that really covers it properly.
I'd be happy if someone found a solution for this though.
